### PR TITLE
test: cleanup target dir tests

### PR DESCRIPTION
**Pull request by rgrinberg**
* remove redundant tests
* split promotion tests into separate file
* add another test that demonstrates a bad error message

### CODE REVIEW

**Comment by rgrinberg** [comment]
@snowleopard ping on this one. it's a pretty simple change, just remove redundant tests and split things to make it easier to run individual tests.

**Comment by snowleopard** [comment]
@rgrinberg Ah, sorry, I got confused when looking at this first time. You say you are deleting redundant tests, but that doesn't seem to be true? I think you are deleting tests for the old way of having directory targets, which didn't really work well but was somehow supported by Dune. I'm not actually sure what we want to do about that feature. Perhaps, deprecate? Turn this into an error?

File: test/blackbox-tests/test-cases/dir-target-dep.t/file-overlap-dir/dune [hunk 1]
```diff
@@ -1,3 +0,0 @@
-(rule
- (targets dir)
```
Comment (snowleopard) [inline test/blackbox-tests/test-cases/dir-target-dep.t/file-overlap-dir/dune:2]: See, it's `(targets dir)` instead of `(targets (dir dir))`.

**Comment by snowleopard** [comment]
@rgrinberg Agreed. We can keep these tests for now, but I think we can eventually drop them and start reporting an error when an action produces a directory target, where a file was declared (giving a hint to use the new syntax for directory targets). Since we never promised to support this feature, it seems like the right way forward.

**Comment by rgrinberg** [comment]
ping @snowleopard I've restored the old tests and merged them into a single parent directory with the other tests.

**Pull request merged**